.github/workflows/publish-qemu-bin.yml

```diff
@@ -0,0 +1,163 @@
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Get current Qemu version
+      id: qemu-version
+      run: |
+        qemuVersion=$(jq -r .linux $QEMU_VERSION_FILE)
+        echo "QEMU_VERSION=$qemuVersion" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Install Dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install --yes --no-install-recommends \
+          autoconf \
+          automake \
+          autotools-dev \
+          bc \
+          build-essential \
+          dosfstools \
+          gcc \
+          libglib2.0-dev \
+          libgtk-3-dev \
+          libpixman-1-dev \
+          libsdl2-dev \
+          mtools \
+          ninja-build \
+          tar
+    - name: Compile Qemu ${{ steps.qemu-version.outputs.QEMU_VERSION }}
+      env:
+        QEMU_VERSION: ${{ steps.qemu-version.outputs.QEMU_VERSION }}
+        QEMU_URL: "https://gitlab.com/qemu-project/qemu.git"
+        DEBIAN_FRONTEND: "noninteractive"
+      run: |
+        git clone $QEMU_URL --branch $QEMU_VERSION --depth 1 qemu
+        cd qemu
+        mkdir bin
+        cd bin
+        ./../configure --target-list=x86_64-softmmu,aarch64-softmmu --enable-gtk
+        sudo make -j $(nproc)
+    - name: Stage Qemu Binaries
+      run: |
+        mkdir -p temp/share
+        cp qemu/bin/qemu-system-x86_64 qemu/bin/qemu-system-aarch64 temp
+        cp qemu/bin/qemu-bundle/usr/local/share/qemu/vgabios-cirrus.bin temp/share
+        cp qemu/bin/qemu-bundle/usr/local/share/qemu/vgabios-stdvga.bin temp/share
+        cp qemu/bin/qemu-bundle/usr/local/share/qemu/kvmvapic.bin temp/share
+    - name: Upload Qemu Artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: qemu-linux
+        path: |
+          temp/*
+        if-no-files-found: error
+  
+  prepare-extdep:
+    name: Prepare and Upload Qemu External Dependency
+    needs: [qemu-windows, qemu-linux]
+    runs-on: Ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        run: |
+          pip install edk2-pytool-extensions
+      - name: Download Windows Qemu Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: qemu-windows
+          path: windows
+      - name: Download Linux Qemu Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: qemu-linux
+          path: linux
+      - name: Stage Ext-Dep
+        run: |
+          mkdir -p qemu
+          mv windows/ qemu/Windows/
+          mv linux/ qemu/Linux/
+          ls qemu/Windows
+          ls qemu/Linux
+          chmod a+x qemu/Linux/*
+      - name: Package Ext-Dep
+        run: |
+          zip -r qemu-${{github.event.release.tag_name}}.zip qemu/*
+          tar -czf qemu-${{github.event.release.tag_name}}.tar.gz qemu/*
+
+      - name: Upload Ext-Dep (Artifact)

```

Comment:
If i understand this correctly, we will only be able to use release versions officially, right? That is there would be no way for a given release tag to use the qemu from its own release as an ext-dep right?

Reply:
Yes, that is correct - It is the same issue we face with BaseTools extdep in MU_BASECORE where only a commit *after* a release, could we start using the extdep from said release.

Reply:
Yeah, I recall that. I don't honest have another solution short of using nuget or something which isn't any better IMO.

Reply:
One solution we could look at running the pipeline when each PR is committed to the default branch, and upload / replace the previous version in the release draft. Once the draft is published, it is locked in.

Reply:
I think the fundamental issue is that a version is being applied to two independent things - (1) firmware code for QEMU (2) QEMU itself.

Ideally, qemu should be released independently and only the version being consumed tracked in this repo.

Reply:
My understanding is we need a better replacement for NuGet feeds.

If we go with the GitHub release attachments the GitHub repo really needs to more closely represent the release (e.g. this repo should just be releasing firmware with the QEMU version tracked).

The workflow could be moved to Mu DevOps as a general catch all for flows like this but then using GitHub releases there would couple it with the Mu DevOps version which is also unrelated to QEMU. The Mu DevOps GitHub NuGet feed is available so the package could be independently versioned but we don't want to use NuGet.

In lieu of that, I was fine with making do here.